cookiejar.go

```diff
@@ -13,32 +13,11 @@ type CookieJar struct {
 	hostCookies map[string][]*Cookie
```

Comment:
maybe hostCookies like this better for reuse memory
```golang
type hc struct {
        key []byte
        value []*Cookie
}
ype CookieJar struct {
    m           sync.Mutex
    hostCookies []hc
}
```

Reply:
I don't think we need to reuse memory in the CookieJar. We are already reusing cookies and I think it's enough. Also, this scheme will speed down the search by host. What do you think?

Reply:
Yes I don't think this will have any impact on allocations for a long running process and it will only complicate the code and make having many hosts slower.

Reply:
Yes, you are right.